rest/config.go

```diff
@@ -676,6 +677,34 @@ func (config *ServerConfig) setupAndValidateDatabases() (errs error) {
 	return nil
 }
 
+// expandEnv replaces $var or ${var} in config according to the values of the
+// current environment variables. The replacement is case-sensitive. References
+// to undefined variables are replaced by an empty string. A default value can
+// be given by using the form ${var:-default value}.
+func expandEnv(config []byte) []byte {
+	return []byte(os.Expand(string(config), func(key string) string {
+		if key == "$" {
+			base.Debugf(base.KeyAll, "Skipping environment variable expansion: %s", key)
+			return key
+		}
+		kvPair := strings.SplitN(key, ":-", 2)
```

Comment:
Along with splitting out this default expansion, and moving the test cases down one level, I'd suggest having a test case which contains `:-` in the actual value itself, to make sure we're only splitting on the first instance in the key.

```golang
func TestEnvDefaultExpansion(t *testing.T) {
	tests := []struct {
		name          string
		envKey        string
		envValue      string
		expectedValue string
	}{
		{
			name:          "no value, no default",
			envKey:        "PASSWORD",
			envValue:      "",
			expectedValue: "",
		},
		{
			name:          "value, no default",
			envKey:        "PASSWORD",
			envValue:      "pa55w0rd",
			expectedValue: "pa55w0rd",
		},
		{
			name:          "value, default",
			envKey:        "PASSWORD:-pa55w0rd",
			envValue:      "foobar",
			expectedValue: "foobar",
		},
		{
			name:          "no value, default",
			envKey:        "PASSWORD:-pa55w0rd",
			envValue:      "",
			expectedValue: "pa55w0rd",
		},
		{
			name:          "no value, default with special chars",
			envKey:        "PASSWORD:-pa55w:-0rd",
			envValue:      "",
			expectedValue: "pa55w:-0rd",
		},
	}

	for _, test := range tests {
		t.Run(test.name, func(t *testing.T) {
			actualValue := envDefaultExpansion(test.envKey, func(s string) string { return test.envValue })
			assert.Equal(t, test.expectedValue, actualValue)
		})
	}
}
```

Reply:
Makes sense, all these tests are included in latest commit.